Some handbook v2 review work and documenting the idea of using typeRoots to type other code (#1307)
microsoft/TypeScript-Website: packages/tsconfig-reference/copy/en/options/rootDirs.md

This does not affect how TypeScript emits JavaScript, it only emulates the assumption that they will be able to
work via those relative paths at runtime.

> `rootDirs` can be used to provide a separate "type layer" to a set of files via `.d.ts` file in another folder, this is a technique useful for bundled applications where you use `import` with files that aren't necessarily code:

Reviewer: ```suggestion
`rootDirs` can be used to provide a separate "type layer" to files that are not TypeScript or JavaScript by providing a home for generated `.d.ts` files in another folder. This is technique is useful for bundled applications where you use `import` of files that aren't necessarily code:
```

```sh
 src
 └── index.ts
 └── css
     └── main.css
     └── navigation.css

 generated
 └── css
     └── main.css.d.ts
     └── navigation.css.d.ts
```

```json tsconfig
{
  "compilerOptions": {
    "rootDirs": ["src", "generated"]
  }
}
```

> Letting you generate types ahead of time for the source files, exclude the resulting types folder and imports work naturally based off the source location.

Reviewer: ```suggestion
This technique lets you generate types ahead of time for the non-code source files.  Imports then work naturally based off the source file's location.
```

> For example the `./src/index.ts` could import the file `./src/css/main.css` and TypeScript can made be aware of the bundler's behavior for that filetype.

Reviewer: ```suggestion
For example `./src/index.ts` can import the file `./src/css/main.css` and TypeScript will be aware of the bundler's behavior for that filetype via the corresponding generated declaration file.
```
